src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -274,7 +276,8 @@ internal static List<LibraryDependency> GetPackageReferences(IMSBuildProject pro
                         string.IsNullOrWhiteSpace(version) ? isCentralPackageVersionManagementEnabled ? null : VersionRange.All : VersionRange.Parse(version),
                         LibraryDependencyTarget.Package),
                     NoWarn = MSBuildStringUtility.GetNuGetLogCodes(packageReferenceItem.GetProperty("NoWarn")).ToList(),
-                    SuppressParent = GetLibraryIncludeFlags(packageReferenceItem.GetProperty("PrivateAssets"), LibraryIncludeFlagUtils.DefaultSuppressParent)
+                    SuppressParent = GetLibraryIncludeFlags(packageReferenceItem.GetProperty("PrivateAssets"), LibraryIncludeFlagUtils.DefaultSuppressParent),
+                    VersionOverride = string.IsNullOrWhiteSpace(versionOverride) ? null : VersionRange.Parse(versionOverride),
```

Comment:
Out of scope, but something to think about, Nuget has really terrible error handling, often crashing/failing restore at first error, rather than trying to limp along to a gatepoint to collect all errors. In this case, if the value for `VersionOverride` can't be parsed as a version range, this project will fail restore immediately, and if there are any other syntax errors, they won't get reported to the customer. They have to fix the errors one by one, retrying restore, getting another single error, then repeat.

Reply:
I saw this too.  At first I was hoping something else happened like the APIs would catch it and present a better error message.  So I tried it out and saw the behavior today if a version is invalid, our code just gives up immediately.  Definitely not the best user experience.  Do we have an existing issue open to fix that?

Reply:
Not that I'm aware of, no. I think it would be great for customers, but I fear how much effort it would be to implement.